### PR TITLE
Add dashboard page

### LINKED ISSUE #39: Create dashboard
Create a dashboard to view a user's forms.

See suggested design here:
https://www.figma.com/file/DOc8uDcOuNze5pdogFKea1/Form-Factory?type=design&node-id=0%3A1&mode=design&t=J4OubVAh5176jrc5-1

### PR DESCRIPTION
**Pull request by ErlingHauan**
This PR adds a dashboard page to the app. The dashboard uses Designsystemet's accordion and modal components to display information about a user's created forms.

When login or signup is successful, the app now redirects to the dashboard.

I also made some small changes to the CSS of the other components, in order to make the design language more coherent.

Closes #39 

https://github.com/ErlingHauan/FormFactory/assets/148075168/7bb6f415-b194-475b-9a08-4f51240d9f30

### CODE REVIEW

File: frontend/main-app/src/components/Dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                <PersonEnvelopeFillIcon />View responses
+              </FormModal>
+              <Button><CloudDownFillIcon />Download data</Button>
+              <Button color="danger"><TrashFillIcon />Delete form</Button>
+            </div>
+          </Accordion.Content>
+        </Accordion.Item>
+      </Accordion>
+    </main>
+  );
```
[the remark this answers is not in the document]
Comment (lassopicasso) [reply frontend/main-app/src/components/Dashboard/Dashboard.tsx:85]: I recommend structuring the dashboard in a way that aligns with how we handle actual data. I've also refactored certain parts to enhance reusability. 
This is just a suggestion for better alignment with our future data needs. Feel free to structure it differently, but I believe taking the future data into consideration would be beneficial. 😃

File: frontend/main-app/src/components/FormModal/FormModal.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+          <Paragraph spacing={true}>Dog</Paragraph>
+          <Divider className={classes.spacing} />
+          <Heading level={2} size="medium" spacing>Response 2</Heading>
+          <Heading level={6} size="small">What is your favorite animal?</Heading>
+          <Paragraph spacing={true}>Dog</Paragraph>
+          <Heading level={6} size="small">What is your 2nd favorite animal?</Heading>
+          <Paragraph spacing={true}>Cat</Paragraph>
+        </Modal.Content>
+      </Modal.Dialog>
+    </Modal.Root>
```
Comment (lassopicasso) [inline frontend/main-app/src/components/FormModal/FormModal.tsx:34]: Perhaps we could also consider how this aligns with actual data, potentially enhancing the reusability of elements, similar to my suggestion for the dashboard. :)

**Review by lassopicasso — CHANGES_REQUESTED** [review]
Well done with this :) I have some comments below.

**Review by lassopicasso — APPROVED** [review]
Well done! 👏 🚀

**Comment by ErlingHauan** [comment]
PR updated! Changes:
* Dashboard now lists all forms that are displayed in a "formData.json" file.
* FormModal now lists all submissions from a  "submittedData.json" file.
* Created a reusable CustomParagraph component.

**Pull request merged**